hazelcast/src/main/java/com/hazelcast/instance/impl/Node.java

```diff
@@ -936,4 +1039,94 @@ private Map<String, String> findMemberAttributes(MemberAttributeConfig attribute
         }
         return attributes;
     }
+
+    @SuppressWarnings({"checkstyle:npathcomplexity", "checkstyle:cyclomaticcomplexity", "checkstyle:methodlength"})
+    @Override
+    public void update(int previousClusterSpecSize, int currentClusterSpecSize,
+                       int readyNodesCount, int currentNodesCount) {
+        if (previousClusterSpecSize == UNKNOWN) {
```

Comment:
is it possible to extract all these new state and intent logic into its own class as much as possible to make it more readable?

Reply:
done in https://github.com/hazelcast/hazelcast/commit/96f2f61ac69d4ac11bcca2b032de9fdfb03269cf